tests/integration/cmd_dev_api_server_test.go

```diff
@@ -63,11 +69,157 @@ var _ = Describe("odo dev command with api server tests", func() {
+						helper.JsonPathExist(strBody, "pid")
+						helper.JsonPathContentIs(strBody, "componentDirectory", commonVar.Context)
+					})
+					By("GETting /component", func() {
+						url := fmt.Sprintf("http://%s/component", devSession.APIServerEndpoint)
+						resp, err := http.Get(url)
+						Expect(err).ToNot(HaveOccurred())
+						Expect(resp.StatusCode).To(BeEquivalentTo(http.StatusOK))
+						defer resp.Body.Close()
+						body, err := io.ReadAll(resp.Body)
+						Expect(err).ToNot(HaveOccurred())
+						strBody := string(body)
+						helper.JsonPathContentIs(strBody, "devfilePath", filepath.Join(commonVar.Context, "devfile.yaml"))
+						helper.JsonPathContentIs(strBody, "devfileData.devfile.metadata.name", cmpName)
+						helper.JsonPathContentIs(strBody, "devfileData.supportedOdoFeatures.dev", "true")
+						helper.JsonPathContentIs(strBody, "devfileData.supportedOdoFeatures.deploy", "false")
+						helper.JsonPathContentIs(strBody, "devfileData.supportedOdoFeatures.debug", "false")
+						helper.JsonPathContentIs(strBody, "managedBy", "odo")
+						if podman {
+							helper.JsonPathDoesNotExist(strBody, "runningOn.cluster")
+							helper.JsonPathExist(strBody, "runningOn.podman")
+							helper.JsonPathContentIs(strBody, "runningOn.podman.dev", "true")
+							helper.JsonPathContentIs(strBody, "runningOn.podman.deploy", "false")
+						} else {
+							helper.JsonPathDoesNotExist(strBody, "runningOn.podman")
+							helper.JsonPathExist(strBody, "runningOn.cluster")
+							helper.JsonPathContentIs(strBody, "runningOn.cluster.dev", "true")
+							helper.JsonPathContentIs(strBody, "runningOn.cluster.deploy", "false")
+						}
+					})
+				})
+			})
+
+			When("odo is executed with --no-watch and --api-server flags", helper.LabelPodmanIf(podman, func() {
+
+				var devSession helper.DevSession
+
+				BeforeEach(func() {
+					var err error
+					args := []string{"--no-watch"}
+					devSession, err = helper.StartDevMode(helper.DevSessionOpts{
+						CmdlineArgs:    args,
+						RunOnPodman:    podman,
+						StartAPIServer: true,
+						EnvVars:        []string{"ODO_EXPERIMENTAL_MODE=true"},
+					})
+					Expect(err).ToNot(HaveOccurred())
+				})
+
+				AfterEach(func() {
+					devSession.Stop()
+					devSession.WaitEnd()
+				})
+
+				When("a file in component directory is modified", func() {
+
+					BeforeEach(func() {
+						helper.ReplaceString(filepath.Join(commonVar.Context, "server.js"), "App started", "App is super started")
+						devSession.CheckNotSynced(10 * time.Second)
+					})
+
+					When("/component/command endpoint is POSTed", func() {
+
+						BeforeEach(func() {
+							url := fmt.Sprintf("http://%s/component/command", devSession.APIServerEndpoint)
+							resp, err := http.Post(url, "application/json", bytes.NewBuffer([]byte(`{"name": "push"}`)))
+							Expect(err).ToNot(HaveOccurred())
+							Expect(resp.StatusCode).To(BeEquivalentTo(http.StatusOK))
+						})
+
+						It("should trigger a push", func() {
```

Comment:
Do you think it makes sense to add this test when `--no-watcher` is _not_ used? Functionally it makes sense to only test it when there is no watcher, but I would still expect it to work just like manual push by pressing `p`.